specs/Environment/AirQualityObserved/harvest/barcelona_airquality_harvest.py

```diff
@@ -207,6 +211,8 @@ def get_air_quality_barcelona(target_stations):
 
     # Now persisting data to Orion Context Broker
     for a_station in entity_data:
+        if a_station not in stations_to_retrieve_data:
```

Comment:
If no station defined, this check will never pass. By default, if no station defined, all available stations should be used.